Makefile

```diff
@@ -94,7 +94,8 @@ tests:
 	go test ./...
 
 vet:
-	go vet ./...
+	go tool vet --all .
+	! go tool vet --shadow . 2>&1 | grep -vE 'declaration of err shadows declaration at (proto/text.go:535|proto/text.go:570|protoc-gen-gogo/main.go:84)'
```

Comment:
Basically the idea would be to run it manually some of the time and decide if the shadows are real bugs.  The next step would be to ONLY fix those that are real bugs.  Since the tool can only provide hints, at this point, as to whether variables are shadowed or not, it can not be incorporated as an automated tool.